metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1717,6 +1719,15 @@ class MetalsLanguageServer(
           scribe.debug(s"Executing AnalyzeStacktrace ${command}")
         }.asJavaObject
 
+      case ServerCommands.Decompile() =>
+        val command = decompiler.decompile(params)
+        command.map { commandOpt =>
+          commandOpt.asScala.foreach(
+            languageClient.metalsExecuteClientCommand
```

Comment:
This should be based on what kind of support a client has (can it display html or json etc.) simialr to how the Doctor works.

Reply:
Assuming that client knows what it supports execute the show client command should be enough - client can handle this request differently, depending on the configuration. 
The response format is taken into account (isHtmlSupported flag)